projects/components/src/gauge/gauge.component.ts

```diff
@@ -0,0 +1,169 @@
+  providers: [SubscriptionLifecycle],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class GaugeComponent implements OnChanges {
+  private static readonly GAUGE_RING_WIDTH: number = 20;
+  private static readonly GAUGE_ARC_CORNER_RADIUS: number = 10;
+
+  @Input()
+  public value?: number;
+
+  @Input()
+  public maxValue?: number;
+
+  @Input()
+  public thresholds: GaugeThreshold[] = [];
+
+  public readonly gaugeRendererData$: Observable<GaugeSvgRendererData>;
+  private readonly inputDataSubject: Subject<GaugeInputData | undefined> = new BehaviorSubject<
+    GaugeInputData | undefined
+  >(undefined);
+  private readonly inputData$: Observable<GaugeInputData | undefined> = this.inputDataSubject.asObservable();
+
+  public constructor(public readonly elementRef: ElementRef) {
+    this.gaugeRendererData$ = this.buildGaugeRendererDataObservable();
+  }
+
+  public ngOnChanges(): void {
+    this.emitInputData();
+  }
+
+  private buildGaugeRendererDataObservable(): Observable<GaugeSvgRendererData> {
+    return combineLatest([this.buildDomResizeObservable(), this.inputData$]).pipe(
+      map(([boundingBox, inputData]) => {
+        const radius = this.buildRadius(boundingBox);
+
+        return {
+          origin: this.buildOrigin(boundingBox, radius),
+          backgroundArc: this.buildBackgroundArc(radius),
+          data: this.buildGaugeData(radius, inputData)
+        };
+      })
+    );
+  }
+
+  private buildDomResizeObservable(): Observable<ClientRect> {
+    const element = this.elementRef.nativeElement as HTMLElement;
+
+    return fromDomResize(element).pipe(debounceTime(200));
+  }
+
+  private buildBackgroundArc(radius: number): string {
+    return this.buildArcGenerator()({
+      innerRadius: radius - GaugeComponent.GAUGE_RING_WIDTH,
+      outerRadius: radius,
+      startAngle: -Math.PI / 2,
+      endAngle: Math.PI / 2
+    })!;
+  }
+
+  private buildGaugeData(radius: number, inputData?: GaugeInputData): GaugeData | undefined {
+    if (inputData === undefined) {
+      return undefined;
+    }
+
+    return {
+      valueArc: this.buildValueArc(radius, inputData),
+      ...inputData
+    };
+  }
+
+  private buildValueArc(radius: number, inputData: GaugeInputData): string {
+    return this.buildArcGenerator()({
+      innerRadius: radius - GaugeComponent.GAUGE_RING_WIDTH,
+      outerRadius: radius,
+      startAngle: -Math.PI / 2,
+      endAngle: -Math.PI / 2 + (inputData.value / inputData.maxValue) * Math.PI
+    })!;
+  }
+
+  private buildArcGenerator(): Arc<unknown, DefaultArcObject> {
+    return arc().cornerRadius(GaugeComponent.GAUGE_ARC_CORNER_RADIUS);
+  }
+
+  private buildRadius(boundingBox: ClientRect): number {
+    return Math.min(boundingBox.height, boundingBox.width) / 2;
+  }
+
+  private buildOrigin(boundingBox: ClientRect, radius: number): Point {
+    return {
+      x: boundingBox.width / 2,
+      y: boundingBox.height / 2 + radius / 2
```

Comment:
This seems strange, assuming a square bounding box, this makes the origin 3/4 of the way down?

Reply:
Yeah. Because we want the center of the gauge to be at the center of the container. Hence the offset by radius/2.

Reply:
So the radius is basically the height of the gauge right? Which means a gauge takes up half the height of its container (why not the full height with some padding) ?

And am I reading this logic right - it's putting the origin y at the bottom of the gauge (not the middle of the gauge), and the origin x in the middle of the gauge (which is always the center of the bounding box).

Reply:
Fixing.

Reply:
done